### PR TITLE
#1646 - Adding Mark Confirmed and Mark Unassigned to Itinerary interface

### PR DESCRIPTION
**Pull request by joro550**
Hoping this can close issue #1646.

### CODE REVIEW

**Comment by tonysurma** [comment]
thank you @joro550 

@mgmccarthy can you take a review?

File: AllReadyApp/Web-App/AllReady/Areas/Admin/Views/Itinerary/Details.cshtml [hunk 1]
```diff
@@ -189,6 +189,9 @@
                             }
                             else if (request.Status == RequestStatus.PendingConfirmation)
                             {
+                                <form asp-action="MarkIncomplete" asp-controller="Itinerary" asp-area="Admin" asp-route-itineraryId="@Model.Id" asp-route-requestId="@request.Id" class="right-action-btn">
```
Comment (mgmccarthy) [inline AllReadyApp/Web-App/AllReady/Areas/Admin/Views/Itinerary/Details.cshtml]: @joro550, We don't want to allow the Request status to move to Incomplete (aka, `RequestStatus.Assigned`) when the request's status is `RequestStatus.PendingConfirmation`. It should only be able to move to `Confirmed` or `Unassigned`. 

so, you can remove this line:
```
<form asp-action="MarkIncomplete" asp-controller="Itinerary" asp-area="Admin" asp-route-itineraryId="@Model.Id" asp-route-requestId="@request.Id" class="right-action-btn">
```

This follows the request status "convention" surrounding what requestors who are sent sms confirmation messages can reply back to us with. For more information, you can check out `SmsResponseController`.

**Comment by joro550** [comment]
@mgmccarthy I have removed the unwanted functionality, thanks for the feedback

**Comment by mgmccarthy** [comment]
@joro550 this looks good. Thanks for you contribution!

@tonysurma :shipit:

**Pull request merged**